airbyte-integrations/connectors/source-zoho-crm/source_zoho_crm/streams.py

```diff
@@ -0,0 +1,153 @@
+        if response.status_code in EMPTY_BODY_STATUSES:
+            return None
+        pagination = response.json()["info"]
+        if not pagination["more_records"]:
+            return None
+        return {"page": pagination["page"] + 1}
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        return next_page_token or {}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        data = [] if response.status_code in EMPTY_BODY_STATUSES else response.json()["data"]
+        yield from data
+
+    def path(self, *args, **kwargs) -> str:
+        return f"/crm/v2/{self.module.api_name}"
+
+    def get_json_schema(self) -> Optional[Dict[Any, Any]]:
+        try:
+            return asdict(self.module.schema)
+        except IncompleteMetaDataException:
+            # to build a schema for a stream, a sequence of requests is made:
+            # one `/settings/modules` which introduces a list of modules,
+            # one `/settings/modules/{module_name}` per module and
+            # one `/settings/fields?module={module_name}` per module.
+            # Any of former two can result in 204 and empty body what blocks us
+            # from generating stream schema and, therefore, a stream.
+            self.logger.warning(f"Could not retrieve fields Metadata for module {self.module.api_name}, skipping")
+            return None
+        except UnknownDataTypeException as exc:
+            self.logger.warning(f"Unknown data type in module {self.module.api_name}, skipping. Details: {exc}")
+            raise
+
+
+class IncrementalZohoCrmStream(ZohoCrmStream):
+    cursor_field = "Modified_Time"
+
+    def __init__(self, authenticator: "requests.auth.AuthBase" = None, config: Mapping[str, Any] = None):
+        super().__init__(authenticator)
+        self._config = config
+        self._state = {}
+        self._start_datetime = self._config.get("start_datetime") or "1970-01-01T00:00:00+00:00"
+
+    @property
+    def state(self) -> Mapping[str, Any]:
+        if not self._state:
+            self._state = {self.cursor_field: self._start_datetime}
+        return self._state
+
+    @state.setter
+    def state(self, value: Mapping[str, Any]):
+        self._state = value
+
+    def read_records(self, *args, **kwargs) -> Iterable[Mapping[str, Any]]:
+        for record in super().read_records(*args, **kwargs):
+            current_cursor_value = datetime.datetime.fromisoformat(self.state[self.cursor_field])
+            latest_cursor_value = datetime.datetime.fromisoformat(record[self.cursor_field])
+            new_cursor_value = max(latest_cursor_value, current_cursor_value)
+            self.state = {self.cursor_field: new_cursor_value.isoformat("T", "seconds")}
+            yield record
+
+    def request_headers(
```

Comment:
do we need to set this header if we are only paginating? e.g: if next_page_token is not none, should this header be set /

Reply:
yes, why not? we are paginating the query set filtered by the header